doc/developers/contributing.rst

```diff
@@ -779,6 +788,56 @@ cross-compatibility and is included in scikit-learn as
 ``sklearn.externals.six``.
 
 
+.. _code_review:
+
+Code Review Guidelines
+======================
+Reviewing code contributed to the project as PRs is a crucial component
+of scikit-learn development. We encourage anyone to start reviewing code of other
+developers. The code review process is often highly educational for everybody involved.
+While each pull request needs to be signed off by two core developers, you can
+speed up this process by providing your feedback.
+
+Here are a few important aspects that need to be covered in any code review,
+from high-level questions to a more detailed check-list.
+
+- Do we want this in the library? Is it likely to be used? Is it in the scope
+  of scikit-learn and does the benefit of the addition warrant the maintenance burden?
+
+- Is the code consistent with the API of scikit-learn? Are public
+  functions/classes/parameters well named and intuitively designed?
+
+- Are all public functions/classes and their parameters, return types, and
+  stored attributes named according to scikit-learn conventions and documented clearly?
+
+- Is any new functionality described in the user-guide and illustrated with examples?
+
+- Is every public function/class tested? Are a reasonable set of parameters,
+  their values, value types, and combinations tested? Do the tests pass in the
+  continuous integration build?
+
+- Do the tests validate that the code is correct, i.e. doing what the
+  documentation says it does?
```

Comment:
If appropriate, help the contributor test the code better.

Maybe should link to some reference on good testing. Often I find contributors struggle with the idea of testing some basic functionality then the effect of individual parameters.